### DISCUSSION TITLE
arduino uno smd board unkown error

### DISCUSSION THREAD

**Discussion by erdemxy** [winder/Universal-G-Code-Sender#2536] [Help] [answered]
I installed grbl on my Arduino Uno - R3 SMD Clone card, I could not connect to the ugs programme, I tried all kinds of grbl versions, baud rates and different version ugs programmes, but it writes unkown when trying to connect to the ugs programme. When I send commands via ardunio ide or python, my stepper motors run smoothly, but I can't connect to the ugs platform, I don't know how to solve it, I thought the problem was due to the fact that I had an arduino uno smd card, but many people on youtubede used the same card and connected without any problems.

**Comment by breiler** [comment]
Make sure that nothing else is connected to the Arduino before connecting with UGS.

Enable verbose logging (right click in the console and activate it). Then try to connect. Paste the console messages here.

**Comment by erdemxy** [comment]
*** Connecting to jssc://COM10:115200
[ayrıntılı] Grbl 1.1h ['$' for help]
Grbl 1.1h ['$' for help]
*** Fetching device status
>>> ?
<Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
<Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
ok
>>> $I
[VER:1.1h.20190825:]
[OPT:V,15,128]
ok
>>> $$
$0 = 10    (Step pulse time, microseconds)
$1 = 25    (Step idle delay, milliseconds)
$2 = 0    (Step pulse invert, mask)
$3 = 0    (Step direction invert, mask)
$4 = 0    (Invert step enable pin, boolean)
$5 = 0    (Invert limit pins, boolean)
$6 = 0    (Invert probe pin, boolean)
$10 = 1    (Status report options, mask)
$11 = 0.010    (Junction deviation, millimeters)
$12 = 0.002    (Arc tolerance, millimeters)
$13 = 0    (Report in inches, boolean)
$20 = 0    (Soft limits enable, boolean)
$21 = 0    (Hard limits enable, boolean)
$22 = 0    (Homing cycle enable, boolean)
$23 = 0    (Homing direction invert, mask)
$24 = 25.000    (Homing locate feed rate, mm/min)
$25 = 500.000    (Homing search seek rate, mm/min)
$26 = 250    (Homing switch debounce delay, milliseconds)
$27 = 1.000    (Homing switch pull-off distance, millimeters)
$30 = 1000    (Maximum spindle speed, RPM)
$31 = 0    (Minimum spindle speed, RPM)
$32 = 0    (Laser-mode enable, boolean)
$100 = 250.000    (X-axis travel resolution, step/mm)
$101 = 250.000    (Y-axis travel resolution, step/mm)
$102 = 250.000    (Z-axis travel resolution, step/mm)
$110 = 500.000    (X-axis maximum rate, mm/min)
$111 = 500.000    (Y-axis maximum rate, mm/min)
$112 = 500.000    (Z-axis maximum rate, mm/min)
$120 = 10.000    (X-axis acceleration, mm/sec^2)
$121 = 10.000    (Y-axis acceleration, mm/sec^2)
$122 = 10.000    (Z-axis acceleration, mm/sec^2)
$130 = 200.000    (X-axis maximum travel, millimeters)
$131 = 200.000    (Y-axis maximum travel, millimeters)
$132 = 200.000    (Z-axis maximum travel, millimeters)
ok
>>> $G
[GC:G0 G54 G17 G21 G90 G94 M5 M9 T0 F0 S0]
ok
*** Connected to GRBL 1.1h
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|WCO:0.000,0.000,0.000>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0|Ov:100,100,100>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
[ayrıntılı] <Idle|MPos:0.000,0.000,0.000|FS:0,0>
**** Connection closed ****

**Reply by erdemxy** [reply]
@breiler Did you get a chance to examine it?

**Reply by breiler** [reply]
That looks perfectly fine to me. Can you post your log file (go to the menu `Tools -> Open log directory` and attach the messages.log here).

**Reply by breiler** [reply]
I am guessing that your settings has become corrupted. 

Shut down UGS and then delete these folders:
```
C:\Users\mustafa\AppData\Roaming\ugsplatform
C:\Users\mustafa\AppData\Local\ugsplatform\Cache\var\cache
```

**Reply by erdemxy** [reply]
I tried and even deleted and reinstalled but sorry nothing changed 

![image](https://github.com/winder/Universal-G-Code-Sender/assets/77491079/0fd6ef18-ad95-4794-8d73-be12e17674a8)

**Reply by breiler** [reply]
In the previous log file that you posted you never connected to the machine. Can you try again and connect to the machine and after that attach the log file?

**Reply by erdemxy** [reply]
I'll send it when I get home after work.

**Reply by erdemxy** [reply]
Unfortunately the same problem persists, is there any other solution you can suggest? @breiler

**Reply by breiler** [reply]
Can I have your log file after you have tried to connect to the machine.

**Reply by breiler** [reply]
That Java version is completely untested so you might run in to a bunch of other problems.

The log file did not contain any messages of you connecting to the machine. Did you connect to the controller like I asked you?

**Reply by erdemxy** [reply]
if the new version is a problem, I could not find where to reinstall the old version ‘ jdk 17.0.8.1+1-jre’

**Reply by erdemxy** [reply]
I connected my card to the machine but it still says ‘unkown’. There was no change in the code in the console.

**Reply by breiler** [reply]
That is why I want the log file so I can troubleshoot why it isn't establishing the connection...

I have no idea if the Java version you've installed will cause problems. I just said that it could - and in that case I would have to troubleshoot two problems instead of just one.

The Java version that UGS uses comes bundled in the UGS zip:
https://github.com/winder/Universal-G-Code-Sender?tab=readme-ov-file#downloads

**Reply by breiler** [reply, marked as answer]
The logs did not reveal anything. Can you try an switch to the connection driver JSerialComm (which is the default on new installations):
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/36cc39a2-1347-45f9-8c50-6b8acdb486f9)

**Reply by erdemxy** [reply]
this method really worked, thank you very much, I am grateful for your help

**Reply by breiler** [reply]
Thanks for your patience, I will remove the JSSC driver in coming releases to prevent other users to encounter this problem.
